### ISSUE TITLE
Default import requires typescript-estree dependents to have esModuleInterop: true in tsconfg

### ISSUE THREAD

**Issue by sandersn** [typescript-eslint/typescript-eslint#1153]
createWatchProgram has [`import ts from 'typescript'`](https://github.com/typescript-eslint/typescript-eslint/blob/d4703e1a97760604cadb4167d5de4a897883119f/packages/typescript-estree/src/create-program/createWatchProgram.ts#L4) which requires `esModuleInterop: true` in the tsconfig of any dependent of typescript-estree.

Typescript doesn't actually have a default export, but a commonjs one. One of these syntaxes would work:

```ts
import * as ts from 'typescript'
// or
import ts = require('typescript')
```

Typescript's nightly user test suite [exposed this problem](https://github.com/microsoft/TypeScript/pull/34759/files), because it compiles prettier with allowJs/checkJs on, but not esModuleInterop.
